addons/hr_holidays/models/hr_leave.py

```diff
@@ -340,6 +361,14 @@ def action_approve(self):
         if any(holiday.state != 'confirm' for holiday in self):
             raise UserError(_('Leave request must be confirmed ("To Approve") in order to approve it.'))
 
+        for holiday in self:
+            validation_type = holiday.holiday_status_id.validation_type
+            department = holiday.employee_id.department_id
+            if (holiday.holiday_status_id.double_validation or validation_type == 'manager') and department.manager_id != current_employee:
+                raise ValidationError('You must be {} manager to approve this leave'.format(department.name))
```

Comment:
Your message should be translated using `_` method.
Also, rather use the `%s` notation to format the message.